csharp/ql/lib/semmle/code/csharp/frameworks/Sql.qll

```diff
@@ -67,18 +68,46 @@ private class IDbCommandConstructionSinkModelCsv extends SinkModelCsv {
         // EntityCommand
         "System.Data.EntityClient;EntityCommand;false;EntityCommand;(System.String);;Argument[0];sql;manual",
         "System.Data.EntityClient;EntityCommand;false;EntityCommand;(System.String,System.Data.EntityClient.EntityConnection);;Argument[0];sql;manual",
-        "System.Data.EntityClient;EntityCommand;false;EntityCommand;(System.String,System.Data.EntityClient.EntityConnection,System.Data.EntityClient.EntityTransaction);;Argument[0];sql;manual"
+        "System.Data.EntityClient;EntityCommand;false;EntityCommand;(System.String,System.Data.EntityClient.EntityConnection,System.Data.EntityClient.EntityTransaction);;Argument[0];sql;manual",
+        // SQLiteCommand
+        "System.Data.SQLite;SQLiteCommand;false;SQLiteCommand;(System.String);;Argument[0];sql;manual",
+        "System.Data.SQLite;SQLiteCommand;false;SQLiteCommand;(System.String,System.Data.SQLite.SQLiteConnection);;Argument[0];sql;manual",
+        "System.Data.SQLite;SQLiteCommand;false;SQLiteCommand;(System.String,System.Data.SQLite.SQLiteConnection,System.Data.SQLite.SQLiteTransaction);;Argument[0];sql;manual",
       ]
   }
 }
 
-/** A construction of an `SqlDataAdapter` object. */
+/** Data flow for SqlCommand and friends. */
+private class SqlCommandSummaryModelCsv extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        // SqlCommand
+        "System.Data.SqlClient;SqlCommand;false;SqlCommand;(System.String);;Argument[0];Argument[Qualifier];taint;manual",
```

Comment:
Why is the output for all constructors `Argument[Qualifier]` and not `ReturnValue`?

Reply:
Constructors are speciel in the way that they are considered to be side effect based, ie. that they manipulate `this`. As far as I remember the reasoning behind this is related to inheritance or maybe I am mistaken?

Reply:
For other constructors, like `Tuple<>`, we use `ReturnValue` (see https://github.com/github/codeql/blob/main/csharp/ql/lib/semmle/code/csharp/frameworks/System.qll#L1323).

Reply:
Oh no - you are right. We need to fix all those manual written summaries - I don't think they work at all. Constructors are considered to be side effect based (ie. they manipulate an implicit this argument).
Fortunately, the model generator already handles constructors correctly.
Should I make a tech debt PR and start to fix all the hand written summaries (we (and by we  - I mean I) must have converted constructors incorrectly when removing the legacy data flow library))?

Reply:
> Should I make a tech debt PR and start to fix all the hand written summaries

Yes, we should definitely fix those broken summaries. Thanks :-)

Reply:
As discussed offline, `Argument[Qualifier]` is arguably the right thing to write. Sorry about the confusion.